src/Utilities/Functional.hpp

```diff
@@ -9,12 +9,67 @@
 
 #include "ErrorHandling/Assert.hpp"
 #include "ErrorHandling/StaticAssert.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Math.hpp"
 
 /*!
  * \ingroup UtilitiesGroup
  * \brief Higher order function objects similar to `std::plus`, etc.
+ *
+ * \details
+ * These chaining function objects can be used to represent highly general
+ * mathematical operations
+ * 1. as types, which can be passed around in template arguments, and
+ * 2. such that any time they can be evaluated at compile time, they will be.
+ *
+ * As an example usecase, consider a function templated on a generic analytic
+ * solution
+ * ```
+ * template <class Sol, class... Ts> sample(Ts... params) { ...
+ * ```
+ * In the corresponding namespace, we could use these functional objects to
+ * define useful distributions, or generate them in the user code.
+ * ```
+ * using Sinusoid = funcl::Multiplies<funcl::Identity,funcl::Sin<
+ *                          funcl::Plus<funcl::Identity, funcl::Multiplies<>>>>
+ * ```
+ * which then gives a function type which when instantiated and evaluated will
+ * give the answer \f$ a*\sin(b + c *d)\f$ from calling `Sinusoid{}(a,b,c,d)`
+ *
+ * As a more creative example, we can take advantage of literals to make, for
```

Comment:
The two examples below are lengthy, slightly overwhelming, and unlikely to represent a realistic use case. We think the sinusoid example is sufficient and the below should be dropped from this documentation.

Reply:
I'm happy to boil the gaussian example down to a more manageable level, but as a significant motivation in creating these was ease in making general tests, I think at least some exposure to the below tests is valuable.

Reply:
Sorry, I don't understand; Did the code snippet come out wrong? It's intended to be a bunch of function calls to a generic function testing a functor type against a lambda, which is intended to be described immediately prior. I had previously included the full function definition previously as well, but @nilsdeppe suggested just the calls were more informative than the function definition as well.

Reply:
the rendered documentation described testing with random values and the snippet appeared as a bunch of calls to a function with no relation to that description.

Reply:
thanks, understood. The documentation was intended to be a description of the function definition now omitted to provide context. I'll trim the calls and rewrite the documentation so it is clearer.